ibis/backends/pyspark/tests/test_timecontext.py

```diff
@@ -55,3 +62,38 @@ def test_table_with_timecontext(client):
 )
 def test_combine_time_context(contexts, expected):
     assert combine_time_context(contexts) == expected
+
+
+def test_adjust_context_scope(client):
+    """Test that `adjust_context` has access to `scope` by default."""
+    table = client.table('time_indexed_table')
+
+    # WindowOp is the only context-adjusted node that the PySpark backend
+    # can compile.
+    # Override its `adjust_context` function with a function that imply
+    # checks that `scope` has been passed in:
+
+    @adjust_context.register(ops.WindowOp)
+    def adjust_context_window_check_scope(
+        op: ops.WindowOp,
+        timecontext: TimeContext,
+        scope: Optional[Scope] = None,
+    ) -> TimeContext:
+        """Confirms that `scope` is passed in."""
+        assert scope is not None
+        return timecontext
+
+    # Do an operation that will trigger context adjustment
+    # on a WindowOp
+    expr = table.mutate(
+        win=table['value'].count().over(
+            ibis.window(
+                ibis.interval(hours=1),
+                0,
+                order_by='time',
+                group_by='key',
+            )
+        )
+    )
+    context = (pd.Timestamp('20170105'), pd.Timestamp('20170111'))
+    expr.execute(timecontext=context)
```

Comment:
I think the assertion on line 83 is probably the one. The results won't really change with and without this PR because the context adjustment rule here doesn't really need "scope" so it's a bit hard to test here. 

I guess we can be fancy and try to inject something into scope and force it to be used in the adjust_context function to change the result, but not sure if it's worthy it.

Reply:
I see. That makes sense.